### PR TITLE
feat: add SpotIM comments to articles on web

### PR DESCRIPTION
**Pull request by Belema**
This PR adds commenting to our article pages. See [JIRA ticket](https://nidigitalsolutions.jira.com/browse/REPLAT-3000).

In the SSR package,
```
yarn bundle:dev
GRAPHQL_ENDPOINT="<TimesPublicApiUrl>" SPOT_ID="<SpotImAccountId>" yarn start
```

### CODE REVIEW

File: packages/article-comments/src/article-comments.web.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  disposeComments() {
+    if (this.container) {
+      this.container.innerHTML = "";
+    }
+  }
+
+  render() {
+    const { commentsEnabled } = this.props;
+    if (!commentsEnabled) {
```
Comment (parkjh) [inline packages/article-comments/src/article-comments.web.js]: can this logic be done by the parent component? rather then having to put this if statement?

File: packages/article-comments/src/comment-login.web.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* eslint-env browser */
+
+const ssoCallback = (codeA, completeSSOCallback) => {
+  const url = `/api/comments/login?codeA=${encodeURIComponent(codeA)}`;
+  const xhr = new XMLHttpRequest();
```
Comment (parkjh) [inline packages/article-comments/src/comment-login.web.js]: do we not use any other xhr request library?

Comment (Belema) [reply packages/article-comments/src/comment-login.web.js]: As discussed, we won't worry about this here/now.

**Comment by parkjh** [comment]
@Belema are there no tests in cypress to assert / expect that the comments exist on the page and that they behave as expected?

**Comment by Belema** [comment]
I am on it @parkjh 👍

File: packages/ssr/__tests__/integration/article.js [hunk 3]
```diff
@@ -3,13 +3,23 @@ import { MockArticle } from "@times-components/fixture-generator";
 const relatedArticleCount = 3;
 
 describe("Article", () => {
+  let SundayTimesArticleWithThreeRelatedArticles;
```
Comment (tuncaulubilge) [inline packages/ssr/__tests__/integration/article.js]: Shouldn't this start with a lowercase?

File: packages/ssr/package.json [hunk 4]
```diff
@@ -13,7 +13,7 @@
     "prepublishOnly": "yarn bundle:prod",
     "prettier:diff": "prettier --list-different '**/*.*'",
     "start": "webpack --config=src/standalone-renderer/webpack.config.js && node src/standalone-renderer/app.js",
-    "start:testservers": "GRAPHQL_ENDPOINT=http://localhost:4000/graphql yarn start & wait-on tcp:3000",
+    "start:testservers": "GRAPHQL_ENDPOINT=http://localhost:4000/graphql SPOT_ID=5p0t_1m_1d yarn start & wait-on tcp:3000",
```
Comment (tuncaulubilge) [inline packages/ssr/package.json]: Feel like we should add SPOT_ID to a README as well, I couldn't see any readme changes on the PR

Comment (tuncaulubilge) [reply packages/ssr/package.json]: SSR readme needs updating

File: packages/article-comments/src/article-comments.web.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  componentDidMount() {
+    this.initialiseComments();
+  }
+
+  componentWillUnmount() {
+    this.disposeComments();
+  }
+
+  initialiseComments() {
+    const { articleId, readOnly, spotAccountId } = this.props;
```
Comment (aronshamash) [inline packages/article-comments/src/article-comments.web.js]: `readOnly` -> `isReadOnly` ?

File: packages/article-comments/src/styles/responsive.web.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+export const CommentContainer = styled.div`
+  margin-left: auto;
+  margin-right: auto;
+
+  @media (min-width: ${breakpoints.medium}px) {
+    width: 80.8%;
+  }
+
+  @media (min-width: ${breakpoints.wide}px) {
+    width: 56.2%;
```
Comment (aronshamash) [inline packages/article-comments/src/styles/responsive.web.js]: we need to put these % width somewhere central but not for this PR.

File: packages/article-comments/src/styles/responsive.web.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  }
+`;
+
+export const CommentGuidelines = styled.p`
+  color: ${colours.functional.secondary};
+  font-family: "${fonts.supporting}";
+  font-size: ${fontSizes.commentsGuidelines}px;
+  margin-bottom: -10px;
```
Comment (aronshamash) [inline packages/article-comments/src/styles/responsive.web.js]: `spacing(2)` ?

File: packages/article-comments/src/styles/responsive.web.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  margin-left: 7px;
+  margin-top: 24px;
```
Comment (aronshamash) [inline packages/article-comments/src/styles/responsive.web.js]: trying to have consistent margins as multiples of our base unit of 5px. Can we use `spacing(5)` ?

File: packages/ssr/package.json (hunk 4 above)
Comment (Belema) [reply packages/ssr/package.json]: Cool 😓

File: packages/article-comments/src/article-comments.web.js (hunk 5 above)
Comment (Belema) [reply packages/article-comments/src/article-comments.web.js]: Ok

**Comment by Belema** [comment]
@tuncaulubilge, @aronshamash, all updated now. Can you have another look?

**Pull request merged**